src/app/authentication/login/login.component.ts

```diff
@@ -37,21 +45,41 @@ export class LoginComponent implements OnInit {
       password:['', Validators.required],
     })
   }
-  buildSignUpFrom(){
+  buildSignUpForm(){
     this.signUpForm = this.formBuilder.group({
+      password:[''],
+      phoneNumber: this.formBuilder.array([this.formBuilder.control('')]),
+      hobbies:[['']],
+      duration:[''],
+      startDate:[''],
+      endDate:[''],
+      interests: [['']],
```

Comment:
you will want to initialise this like you did for Phone number. now to accomplish this you may have to statically do the "this.formBuilder.control('')" for the array or dynamically iterate through your interests array to accomplish this